### PR TITLE
returning_progress_in_the_end

### PR DESCRIPTION
**Pull request by Ilya-Krylov**

### CODE REVIEW

File: mmdet/apis/ote/extension/utils/hooks.py [hunk 1]
```diff
@@ -131,7 +131,7 @@ def before_run(self, runner):
         self.time_monitor.total_epochs = total_epochs
         self.time_monitor.train_steps = runner.max_iters // total_epochs
         self.time_monitor.steps_per_epoch = self.time_monitor.train_steps + self.time_monitor.val_steps
-        self.time_monitor.total_steps = math.ceil(self.time_monitor.steps_per_epoch * total_epochs)
+        self.time_monitor.total_steps = max(math.ceil(self.time_monitor.steps_per_epoch * total_epochs), 1)
```
Comment (druzhkov-paul) [inline mmdet/apis/ote/extension/utils/hooks.py:134]: How is that possible that the result is < 1?

File: tests/test_ote_api.py [hunk 2]
```diff
@@ -285,16 +285,24 @@ def test_cancel_training_detection(self):
...
-        train_future = executor.submit(detection_task.train, dataset, output_model)
+        train_future = executor.submit(detection_task.train, dataset, output_model, train_parameters)
         # give train_thread some time to initialize the model
         while not detection_task._is_training:
             time.sleep(10)
         detection_task.cancel_training()
 
         # stopping process has to happen in less than 35 seconds
         train_future.result()
+        assert training_progress_curve[-1] == 100
```
Comment (druzhkov-paul) [inline tests/test_ote_api.py]:
```suggestion
        self.assertEqual(training_progress_curve[-1], 100)
```

Comment (Ilya-Krylov) [reply tests/test_ote_api.py]: >       assert training_progress_curve[-1] == 101
E       AssertionError: assert 100.0 == 101
E         +100.0
E         -101

>       self.assertEqual(training_progress_curve[-1], 101)
E       AssertionError: 100.0 != 101

What do you like more?

File: mmdet/apis/ote/extension/utils/hooks.py (hunk 1 above)
Comment (Ilya-Krylov) [reply mmdet/apis/ote/extension/utils/hooks.py:134]: Yes, our optimization partners said that sometimes their system tells that there is no need to do any fine-tuning step after optimization, so train_steps = 0. But 'after_epoch' callback is called anyway [here](https://github.com/open-mmlab/mmcv/blob/master/mmcv/runner/epoch_based_runner.py#L130) that causes division by 0 while computing progress.

File: tests/test_ote_api.py (hunk 2 above)
Comment (druzhkov-paul) [reply tests/test_ote_api.py]: The latter one.

**Review by druzhkov-paul — APPROVED** [review]

**Pull request merged**